managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -141,7 +141,11 @@ public void safeRun() {
         ManagedLedgerImpl.TOTAL_SIZE_UPDATER.addAndGet(ml, dataLength);
         if (ml.hasActiveCursors()) {
             // Avoid caching entries if no cursor has been created
-            ml.entryCache.insert(new EntryImpl(ledger.getId(), entryId, data));
+            EntryImpl entry = EntryImpl.create(ledger.getId(), entryId, data);
+            // EntryCache.insert: duplicates entry by allocating new entry and data. so, recycle entry after calling
+            // insert
+            ml.entryCache.insert(entry);
+            entry.recycle();
```

Comment:
I think as per documentation the reason is buffer coming from netty is in 64kb chunk so, to avoid storing unnecessary memory for smaller entry < 64kb, we copy and store it. So, I think we should change definition of insert as `insert(ByteBuf)` as we don't use passed `EntryImpl` object in method at all. any thought?